disnake/poll.py

```diff
@@ -0,0 +1,415 @@
+            The member (if retrievable) or the user that has voted
+            for this answer. The case where it can be a :class:`Member` is
+            in a guild message context. Sometimes it can be a :class:`User`
+            if the member has left the guild.
+        """
+        if not (self.id is not None and self.poll and self.poll.message):
+            raise ValueError(
+                "This object was manually built. To use this method, you need to use a poll object retrieved from the Discord API."
+            )
+
+        return PollAnswerIterator(self.poll.message, self.id, limit=limit, after=after)
+
+
+class Poll:
+    """Represents a poll from Discord.
+
+    .. versionadded:: 2.10
+
+    Parameters
+    ----------
+    question: Union[:class:`str`, :class:`PollMedia`]
+        The question of the poll. Currently, emojis are not supported in poll questions.
+    answers: List[Union[:class:`str`, :class:`PollAnswer`]]
+        The answers for this poll, up to 10.
+    duration: :class:`datetime.timedelta`
+        The total duration of the poll, up to 32 days. Defaults to 1 day.
+        Note that this gets rounded down to the closest hour.
+    allow_multiselect: :class:`bool`
+        Whether users will be able to pick more than one answer. Defaults to ``False``.
+    layout_type: :class:`PollLayoutType`
+        The layout type of the poll. Defaults to :attr:`PollLayoutType.default`.
+
+    Attributes
+    ----------
+    message: Optional[:class:`Message`]
+        The message which contains this poll. This will be ``None`` only if this object was created manually
+        and did not originate from the API.
+    question: :class:`PollMedia`
+        The question of the poll.
+    duration: Optional[:class:`datetime.timedelta`]
+        The original duration for this poll. ``None`` if the poll is a non-expiring poll.
+    expires_at: Optional[:class:`datetime.datetime`]
+        The date when this poll will expire. ``None`` if the poll is a non-expiring poll
+        or if this object was created manually and did not originate from the API.
+    allow_multiselect: :class:`bool`
+        Whether users are able to pick more than one answer.
+    layout_type: :class:`PollLayoutType`
+        The type of the layout of the poll.
+    is_finalized: :class:`bool`
+        Whether the votes have been precisely counted.
+    """
+
+    __slots__ = (
+        "_state",
+        "message",
+        "question",
+        "_answers",
+        "duration",
+        "expires_at",
+        "allow_multiselect",
+        "layout_type",
+        "is_finalized",
+    )
+
+    def __init__(
+        self,
+        question: Union[str, PollMedia],
+        *,
+        answers: List[Union[str, PollAnswer]],
+        duration: timedelta = timedelta(hours=24),
+        allow_multiselect: bool = False,
+        layout_type: PollLayoutType = PollLayoutType.default,
+    ) -> None:
+        self._state: Optional[ConnectionState] = None
```

Comment:
If I'm not mistaken, usages of `self._state` could be replaced with `self.message._state`.